### ISSUE TITLE
Rust Analyzer incorrectly marks src/ou_process.rs as unlinked

### ISSUE THREAD

**Issue by orzAtalod** [rust-lang/rust-analyzer#20791]
Description:
Rust Analyzer reports that `ou_process.rs` is "not included anywhere in the module tree" and greys it out,
even though `main.rs` contains:

```rust
mod ou_process;

fn main() {
    let foobar = ou_process::OUState::new();
    println!("{:?}", foobar);
}
```

Project structure:

causal-aquarium/
├─ Cargo.toml
└─ src/
    ├─ main.rs
    └─ ou_process.rs

Steps tried:

Added the file to GitHub so that it could be tracked.
Restarted Rust Analyzer multiple times.
Verified Cargo.toml is at workspace root.
Confirmed mod ou_process; is correct.
Restart the computer.

Environment:

Rust Analyzer: 0.3.2593
VSCode: 1.104.3
OS: Windows 11

Expected behavior:

Rust Analyzer should recognize ou_process.rs as part of the module tree and provide IDE features.

Actual behavior:

ou_process.rs is greyed out (To get rid of the grayed-out code, I had to set "editor.showUnused": false.)
Warning: "this file is not included anywhere in the module tree"
Compiler can still build and run the project successfully

Additional info:

I completely deleted my local project and cloned it again from GitHub. After this, the issue was resolved. 
I tried multiple times but was not able to reliably reproduce the problem from a fresh clone or from scratch.

Screenshot:

<img width="1618" height="406" alt="Image" src="https://github.com/user-attachments/assets/a8751c97-7ebe-4ddf-a60a-c1d0601459d0" />

<img width="1338" height="696" alt="Image" src="https://github.com/user-attachments/assets/df35fd46-73fb-48ae-9fda-5c08b7b5a56e" />

**Comment by lnicola** [comment]
Did you rename it from OU_process or something else with upper-case?

**Comment by orzAtalod** [comment]
> Did you rename it from OU_process or something else with upper-case?

Yes I did
